src/abstractdataframe/io.jl

```diff
@@ -21,87 +68,27 @@ function digitsep(value::Integer)
     return join(groups, ',')
 end
 
-function printtable(io::IO,
-                    df::AbstractDataFrame;
-                    header::Bool = true,
-                    separator::Char = ',',
-                    quotemark::Char = '"',
-                    missingstring::AbstractString = "missing")
-    n, p = size(df)
-    etypes = eltypes(df)
-    if header
-        cnames = _names(df)
-        for j in 1:p
-            print(io, quotemark)
-            print(io, cnames[j])
-            print(io, quotemark)
-            if j < p
-                print(io, separator)
-            else
-                print(io, '\n')
-            end
-        end
-    end
-    quotestr = string(quotemark)
-    for i in 1:n
-        for j in 1:p
-            if !ismissing(df[j][i])
-                if ! (etypes[j] <: Real)
-                    print(io, quotemark)
-                    escapedprint(io, df[i, j], quotestr)
-                    print(io, quotemark)
-                else
-                    print(io, df[i, j])
-                end
-            else
-                print(io, missingstring)
-            end
-            if j < p
-                print(io, separator)
-            else
-                print(io, '\n')
-            end
-        end
-    end
-    return
-end
-
-function printtable(df::AbstractDataFrame;
-                    header::Bool = true,
-                    separator::Char = ',',
-                    quotemark::Char = '"',
-                    missingstring::AbstractString = "missing")
-    printtable(stdout,
-               df,
-               header = header,
-               separator = separator,
-               quotemark = quotemark,
-               missingstring = missingstring)
-    return
-end
-##############################################################################
-#
-# HTML output
-#
-##############################################################################
-
 function html_escape(cell::AbstractString)
     cell = replace(cell, "&"=>"&amp;")
     cell = replace(cell, "<"=>"&lt;")
     cell = replace(cell, ">"=>"&gt;")
     return cell
 end
 
-Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame; summary::Bool=true) =
-    _show(io, mime, df, summary=summary)
-
 function _show(io::IO, ::MIME"text/html", df::AbstractDataFrame;
                summary::Bool=true, rowid::Union{Int,Nothing}=nothing)
-    n = size(df, 1)
-    if rowid !== nothing && n != 1
+    if rowid !== nothing && size(df, 1) != 1
         throw(ArgumentError("rowid may be passed only with a single row data frame"))
     end
-    cnames = _names(df)
+
+    mxrow, mxcol = size(df)
+    if get(io, :limit, true)
+        tty_rows, tty_cols = get(io, :displaysize, displaysize(io))
+        mxrow = min(mxrow, tty_rows)
+        mxcol = min(mxcol, tty_cols)
```

Comment:
This is going to print much more columns than at the REPL. Would there be any way to use the width computed by the text/plain `show`? Or maybe divide this by a reasonable factor (e.g. 6-10)?

Reply:
Done, based on actual column widths + 2 characters for padding.

Reply:
Great!